openmmtools/forcefactories.py

```diff
@@ -174,6 +174,366 @@ def restrain_atoms(thermodynamic_state, sampler_state, restrained_atoms, sigma=3
     thermodynamic_state.system = system
 
 
+# ==================== CUSTOM LENNARD-JONES POTENTIALS AND SWITCHING FUNCTIONS ==========================
+
+"""
+Attributes
+----------
+LJ_POTENTIAL: str
+    OpenMM energy expression for a standard 6-12 LJ potential in terms of A and B:
+    "A/r^12 - B/r^6;"
+A_B_FROM_EPSILON_SIGMA: str
+    OpenMM energy expression for calculating A and B in the LJ potential from sigma and epsilon:
+    "A = 4.0*epsilon*sigma^12; B = 4.0*epsilon*sigma^6;"
+A_B_FROM_ACOEF_BCOEF: str
+    OpenMM energy expression for calculating A and B from the acoef and bcoef fields that
+    define NBFIXes in openmm.app.CharmmPsfFile.createSystem:
+    "A = acoef(type1, type2)^2; B = bcoef(type1, type2);"
+LORENTZ_BERTEHLOT_RULES: str
+    OpenMM energy expression for calculating sigma and epsilon by Lorentz-Berthelot mixing rules
+    (geometric mean for epsilon, arithmetic mean for sigma):
+    "epsilon=sqrt(epsilon1*epsilon2); sigma=0.5*(sigma1+sigma2)"
+CHARMM_VSWITCH_ENERGY_EXPRESSION: str
+    OpenMM energy expressions for the VSWITCH switching function that is used in CHARMM
+    (Steinbach and Brooks, JCC 15 (7), pp. 667-683, 1994):
+    "(cutoff^2 - r^2)^2 * (cutoff^2 + 2*r^2 - 3*switch^2) / (cutoff^2 - switch^2)^3"
+"""
+LJ_POTENTIAL = "A/r^12 - B/r^6;"
+A_B_FROM_EPSILON_SIGMA = "A = 4.0*epsilon*sigma^12; B = 4.0*epsilon*sigma^6;"
+A_B_FROM_ACOEF_BCOEF = "A = acoef(type1, type2)^2; B = bcoef(type1, type2);"
+LORENTZ_BERTHELOT_RULES = "epsilon=sqrt(epsilon1*epsilon2); sigma=0.5*(sigma1+sigma2)"
+CHARMM_VSWITCH_ENERGY_EXPRESSION = "(cutoff^2 - r^2)^2 * (cutoff^2 + 2*r^2 - 3*switch^2) / (cutoff^2 - switch^2)^3"
+
+
+def is_lj_active_in_nonbonded_force(nonbonded_force):
+    """Check if the LJ potential is defined in the openmm.NonbondedForce or if all terms are zero.
+
+    Parameters
+    ----------
+    nonbonded_force : openmm.NonbondedForce
+        The nonbonded force of a openmm.System.
+
+    Returns
+    -------
+    is_lj_active : bool
+    """
+    for i in range(nonbonded_force.getNumParticles()):
+        q, sig, eps = nonbonded_force.getParticleParameters(i)
+        vdw = (eps.value_in_unit(unit.kilojoule_per_mole) * sig.value_in_unit(unit.nanometer))
+        if vdw > 1e-5:
+            return True
+    for i in range(nonbonded_force.getNumExceptions()):
+        particle1, particle2, qq, sig, eps = nonbonded_force.getExceptionParameters(i)
+        vdw = (eps.value_in_unit(unit.kilojoule_per_mole) * sig.value_in_unit(unit.nanometer))
+        if vdw > 1e-5:
+            return True
+    return False
+
+
+def get_forces_that_define_vdw(system):
```

Comment:
Maybe rename this `find_vdw_forces` for consistency with `forces.find_forces`?

Reply:
It looks like this function encapsulates some logic that is also exposed in `alchemy.is_nbfix_force` and `is_custom_lj_force`. Does it make sense to remove the redundant logic?